mergify_engine/actions/merge_base.py

```diff
@@ -171,22 +152,76 @@ async def _merge(
                     f"Please make sure `{merge_bot_account}` has logged in Mergify dashboard.",
                 )
 
-        try:
-            await ctxt.client.put(
-                f"{ctxt.base_url}/pulls/{ctxt.pull['number']}/merge",
-                oauth_token=github_user["oauth_access_token"] if github_user else None,
-                json=data,
+        if method == "fast-forward":
+            try:
+                await merge_fast_forwarder.merge(ctxt, github_user)
+            except merge_fast_forwarder.MergeFastForwardBaseMoved:
+                return await self.get_queue_status(ctxt, rule, q)
+            except merge_fast_forwarder.MergeFastForwardFailed as e:
+                message = "Mergify failed to merge the pull request"
+                ctxt.log.info(
+                    "fast-foward merge fail",
+                    mergify_message=message,
+                    error_message=e.reason,
+                )
+                return check_api.Result(
+                    check_api.Conclusion.FAILURE,
+                    message,
+                    f"GitHub error message: `{e.reason}`",
+                )
+
+            await self.send_signal(ctxt)
+            # NOTE(sileht): We can't use merge_report() here, because it takes
+            # some times for GitHub to detect this pull request has been
+            # merged. Just after the fastforward git push, mergeable_state is
+            # mark as conflict and a bit later as unknown and merged attribute set to
+            # true. We can't block here just for this.
```

Comment:
```suggestion
            # NOTE(sileht): We can't use merge_report() here, because it takes
            # some time for GitHub to detect this pull request has been
            # merged. Just after the fast-forward git push, mergeable_state is
            # mark as conflict and a bit later as unknown and merged attribute set to
            # true. We can't block here just for this.
```